### PR TITLE
Remove unused args from functions

### PR DESCRIPTION
**Pull request by compiler-errors**
`#[instrument]` suppresses the unused arguments from a function, *and* suppresses unused methods too! This PR removes things which are only used via `#[instrument]` calls, and fixes some other errors (privacy?) that I will comment inline.

It's possible that some of these arguments were being passed in for the purposes of being instrumented, but I am unconvinced by most of them.

### CODE REVIEW

File: compiler/rustc_query_system/src/dep_graph/graph.rs [hunk 1]
```diff
@@ -52,7 +52,7 @@ impl From<DepNodeIndex> for QueryInvocationId {
     }
 }
 
-pub(crate) struct MarkFrame<'a> {
+pub struct MarkFrame<'a> {
```
Comment (compiler-errors) [inline compiler/rustc_query_system/src/dep_graph/graph.rs:55]: We started getting a private-in-public error when the instrument call was removed from `try_mark_parent_green`.

**Review by Noratrieb — APPROVED** [review]
very cool. :( that tracing stops the unused vars, i wish there was a way to mark uses as not used

**Comment by matthiaskrgr** [comment]
@bors r-
failed here I think https://github.com/rust-lang/rust/pull/120651#issuecomment-1926063178

**Comment by compiler-errors** [comment]
lmao this pr has nothing to do with that rollup failure

@bors r=Nilstrieb

**Pull request merged**